physics/grahams_law.py

```diff
@@ -0,0 +1,211 @@
+"""
+Title: Graham's Law of Effusion
+
+Description: Graham's law of effusion states that the rate of effusion of a gas is
+inversely proportional to the square root of the molar mass of its particles:
+
+r1/r2 = sqrt(m2/m1)
+
+r1 = Rate of effusion for the first gas.
+r2 = Rate of effusion for the second gas.
+m1 = Molar mass of the first gas.
+m2 = Molar mass of the second gas.
+
+(Description adapted from https://en.wikipedia.org/wiki/Graham%27s_law)
+"""
+
+from math import pow, sqrt
+
+
+def validate(*values: float) -> bool:
+    """
+    Input Parameters:
+    -----------------
+    effusion_rate_1: Effustion rate of first gas (m^2/s, mm^2/s, etc.)
+    effusion_rate_2: Effustion rate of second gas (m^2/s, mm^2/s, etc.)
+    molar_mass_1: Molar mass of the first gas (g/mol, kg/kmol, etc.)
+    molar_mass_2: Molar mass of the second gas (g/mol, kg/kmol, etc.)
+
+    Returns:
+    --------
+    >>> validate(2.016, 4.002)
+    True
+    >>> validate(-2.016, 4.002)
+    Traceback (most recent call last):
+      ...
+    ValueError: Invalid inputs. Effusion rates and molar masses must be a positive
+                    value.
+    >>> validate()
+    False
+    """
+    for value in values:
+        if value >= 0.0:
+            return True
+        else:
+            raise ValueError(
+                """Invalid inputs. Effusion rates and molar masses must be a positive
+                value."""
+            )
+
+    return False
+
+
+def effusion_ratio(molar_mass_1: float, molar_mass_2: float) -> float:
+    """
+    Input Parameters:
+    -----------------
+    molar_mass_1: Molar mass of the first gas (g/mol, kg/kmol, etc.)
+    molar_mass_2: Molar mass of the second gas (g/mol, kg/kmol, etc.)
+
+    Returns:
+    --------
+    >>> effusion_ratio(2.016, 4.002)
+    1.408943
+    >>> effusion_ratio(-2.016, 4.002)
```

Comment:
Please add a test for
```
    >>> effusion_ratio(2.016, -4.002)
```
and explain why it returns the wrong result.

Python's builtin `all()` or `any()` might be a better fit.
* https://docs.python.org/3/library/functions.html#all